collect/collect.go

```diff
@@ -235,6 +248,7 @@ func (i *InMemCollector) checkAlloc() {
 	// To do this, we sort the traces by their CacheImpact value and then remove traces
 	// until the total size is less than the amount to which we want to shrink.
 	allTraces := i.cache.GetAll()
+	cacheSize := len(allTraces)
```

Comment:
I think here it makes more sense to just take the length of all returned traces since the `Len()` function from the new cache requires grabbing a lock first